hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/bootstrap/BootstrapUtils.java

```diff
@@ -79,7 +79,7 @@ public static List<Pair<String, List<HoodieFileStatus>>> getAllLeafFoldersWithFi
       result.addAll(context.flatMap(subDirectories, directory -> {
         PathFilter pathFilter = getFilePathFilter(baseFileExtension);
         Path path = new Path(directory);
-        FileSystem fileSystem = path.getFileSystem(new Configuration());
+        FileSystem fileSystem = FSUtils.getFs(path, fs.getConf());
         RemoteIterator<LocatedFileStatus> itr = fileSystem.listFiles(path, true);
```

Comment:
`fs` is already a `FileSystem`, why new an instance here ?

Reply:
For context.flatMap is not thread safe, three is many thread in `context.flatMap` may operate same `fs` if not create new one in `directory -> {} function`.

Reply:
-> may operate same fs

What's the problem here, the fs instance is cached BTW, so you would still fetch the same fs instance each time.

Reply:
So we just remove `path.getFileSystem(new Configuration());`, and rename `fs` ouside to `filesystem` in lambda，right?

Reply:
I guess it though ~

Reply:
@danny0405 Hi, I have fixed it, please help review again ~

Reply:
What bug is fixed here then ? Or simplify a code refactoring.